src/datasets/teacher.py

```diff
@@ -0,0 +1,32 @@
+"""Create teacher dataset for interactive learning.
+"""
+
+from typing import List, Set
+import numpy as np
+from predicators.src.envs import BaseEnv
+from predicators.src.structs import Dataset, GroundAtom
+from predicators.src.settings import CFG
+from predicators.src import utils
+
+
+def create_teacher_dataset(env: BaseEnv,
```

Comment:
one more very minor thing: maybe we should pass in predicates here instead of `env`, since the only thing from env that we're using is `env.predicates`? but I know that the other dataset functions take in `env`, so maybe it's nice to be parallel. I'll leave this up to you!

Reply:
this function is already different from the other dataset functions, so I think it is ok to pass in just the predicates!